src/Development/IDE/GHC/Compat.hs

```diff
@@ -178,3 +187,66 @@ dontWriteHieFiles d =
 #else
     d
 #endif
+
+nameListFromAvails :: [AvailInfo] -> [(SrcSpan, Name)]
+nameListFromAvails as =
+  map (\n -> (nameSrcSpan n, n)) (concatMap availNames as)
+
+#if !MIN_GHC_API_VERSION(8,8,0)
+-- Reimplementations of functions for HIE files for GHC 8.6
+
+mkHieFile :: ModSummary -> TcGblEnv -> RenamedSource -> ByteString -> Hsc HieFile
```

Comment:
Question for @mpickering

Reply:
I think the two arguments for using `.hie` files are.

1. Other tools might want to use `.hie` files as well, for example, if `weeder` was integrated into `ghcide`.
2. Contributes will not be surprised to find our custom implementation.

Reply:
1 seems compelling. I don’t really buy 2 atm as we have to copy much more code into `ghcide` because the upstream API doesn’t work for us but hopefully that will be fixed in future GHC versions.

Alright, then let’s go with this :+1: